pyccl/halos/profiles.py

```diff
@@ -843,3 +844,214 @@ def _real(self, cosmo, r, M, a, mass_def):
         if np.ndim(M) == 0:
             prof = np.squeeze(prof, axis=0)
         return prof
+
+
+class HaloProfilePressureGNFW(HaloProfile):
+    """ Generalized NFW pressure profile by Arnaud et al.
+    (2010A&A...517A..92A).
+
+    The parametrization is:
+
+    .. math::
+       P_e(r) = C\\times P_0 h_{70}^E (c_{500} x)^{-\\gamma}
+       [1+(c_{500}x)^\\alpha]^{(\\gamma-\\beta)/\\alpha},
+
+    where
+
+    .. math::
+       C = 1.65\\,h_{70}^2\\left(\\frac{H(z)}{H_0}\\right)^{8/3}
+       \\left[\\frac{h_{70}\\tilde{M}_{500}}
+       {3\\times10^{14}\\,M_\\odot}\\right]^{2/3+0.12},
+
+    :math:`x = r/\\tilde{r}_{500}`, :math:`h_{70}=h/0.7`, and the
+    exponent :math:`E` is -1 for SZ-based profile normalizations
+    and -1.5 for X-ray-based normalizations. The biased mass
+    :math:`\\tilde{M}_{500}` is related to the true overdensity
+    mass :math:`M_{500}` via the mass bias parameter :math:`(1-b)`
+    as :math:`\\tilde{M}_{500}=(1-b)M_{500}`. :math:`\\tilde{r}_{500}`
+    is the overdensity halo radius associated with :math:`\\tilde{M}_{500}`
+    (note the intentional tilde!), and the profile is defined for
+    a halo overdensity :math:`\\Delta=500` with respect to the
+    critical density.
+
+    The default arguments (other than `mass_bias`), correspond to the
+    profile parameters used in the Planck 2013 (V) paper.
+
```

Comment:
Could you add the units of the pressure profile here as well for completeness?

Reply:
Good point. Done!